crates/sui-types/src/messages.rs

```diff
@@ -437,6 +444,37 @@ where
         &self.gas_payment
     }
 
+    pub fn move_calls(&self) -> SuiResult<Vec<&MoveCall>> {
+        let move_calls = match &self.kind {
+            TransactionKind::Single(s) => {
+                let mut result = vec![];
+                let sub = s.move_call();
+                if sub.is_some() {
```

Comment:
Instead of all these lines, can do `let result: Vec<_> = s.move_call().iter().collect();`  (An option can be turned into a collection; None => empty collection while Some => collection of 1).